apps/core/declarativewebcontainer.cpp

```diff
@@ -540,6 +540,29 @@ int DeclarativeWebContainer::requestTabWithOwner(int tabId, const QString &url,
     return tabId;
 }
 
+void DeclarativeWebContainer::requestTabWithOwnerAsync(int tabId, const QString &url, uint ownerPid, void *context)
+{
+    // We should only create or activate tabs once the model has loaded
+    if (m_model->loaded()) {
+        // The tab model has already loaded, so we can go ahead and create the tab
+        int activatedTab = requestTabWithOwner(tabId, url, ownerPid);
+        emit requestTabWithOwnerAsyncResult(activatedTab, context);
+    } else {
+        // The model has yet to load, so queue creation of the tab
+        QMetaObject::Connection * const connection = new QMetaObject::Connection;
+        *connection = connect(m_model.data(), &DeclarativeTabModel::loadedChanged, [this, tabId, url, ownerPid, context, connection]() {
```

Comment:
Oh, one comment: could and should this take `this` as context?
https://doc.qt.io/archives/qt-5.6/qobject.html#connect-5

Reply:
What's your concern in particular? The instance needs to be available to emit the signal.

Reply:
Ah. Now I see what you mean. The model instance won't exist if `this` has been destroyed.